collections/userEvents.coffee

```diff
@@ -61,3 +62,13 @@ Meteor.methods
         lastModifiedByUserId: user._id,
         lastModifiedByUserName: user.profile.name
       }})
+  updateUserEventArticleCount: (id, countModifier) ->
+    event = UserEvents.findOne(id)
+    UserEvents.update(id, {$set: {articleCount: event.articleCount + countModifier}})
+
+# Add article count field to events
+if Meteor.isServer
+  Meteor.startup ->
+    UserEvents.find({articleCount: {$exists: false}}).forEach (event) ->
```

Comment:
I would remove filter query and just update the counts on all the events incase one gets out of sync somehow.